tests/binary_tests/flaskapp.test.js

```diff
@@ -122,3 +122,61 @@ test('write parameters to python script', async () => {
   }
   fs.unlinkSync(filepath);
 });
+
+test('validate the UI spec', async () => {
+  const models = [
+    "carbon", 
+    "coastal_blue_carbon",
+    "coastal_blue_carbon_preprocessor",
+    "coastal_vulnerability",
+    "crop_production_percentile",
+    "crop_production_regression",
+    "delineateit",
+    "finfish_aquaculture",
+    "fisheries",
+    "fisheries_hst",
+    "forest_carbon_edge_effect",
+    "globio",
+    "habitat_quality",
+    "habitat_risk_assessment",
+    "hydropower_water_yield",
+    "ndr",
+    "pollination",
+    "recreation",
+    "routedem",
+    "scenario_generator_proximity",
+    "scenic_quality",
+    "sdr",
+    "seasonal_water_yield",
+    "urban_cooling_model",
+    "urban_flood_risk_mitigation",
+    "wave_energy",
+    "wind_energy"
+  ];
+  const uiSpec = require('../../src/ui_config');
+  // get the args spec for each model
+  const argsSpecs = await Promise.all(models.map(
+    async model => await server_requests.getSpec(model)
+  ));
+
+  argsSpecs.forEach((argsSpec) => {
+    // expect each arg in the UI spec to exist in the args spec
+    for (const property in uiSpec[argsSpec.model_name]) {
+      if (property === 'order') {
+        // 'order' is a 2D array of arg names
+        const order_array = uiSpec[argsSpec.model_name].order.flat();
+        const order_set = new Set(order_array);
+        // expect there to be no duplicated args in the order
+        expect(order_array.length).toEqual(order_set.size);
+        order_array.forEach(arg => {
+            expect(argsSpec.args[arg]).toBeDefined();
+        });
+      } else {
+        // for other properties, each key is an arg
+        Object.keys(uiSpec[argsSpec.model_name][property]).forEach(arg => {
+          expect(argsSpec.args[arg]).toBeDefined();
+        });
+      }
+    }
+  });
+});

```

Comment:
yes it would, I think we want that because each model needs an `order` property for its args to be rendered. I just added another `expect` to explicitly check that there is an `order` property.